### PR TITLE
Data Module: update the resolvers to rely on controls instead of async generators

### PR DESCRIPTION
**Pull request by youknowriad**
Follow up to #9507 

This PR refactors the current Gutenberg resolvers to use controls instead of async generator.

**Testing instructions**

 - Unit and e2e tests pass.

**Todo**

 - [x] Deprecate the async generators plugin

### CODE REVIEW

File: packages/core-data/src/controls.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	};
+}
+
+const controls = {
+	API_FETCH( { request } ) {
+		return triggerApiFetch( request );
+	},
+
+	SELECT( { selectorName, args } ) {
+		return selectData( 'core' )[ selectorName ]( ...args );
```
Comment (youknowriad) [inline packages/core-data/src/controls.js]: Ideally, this would rely on a given `registry` object as an argument in the control instead of calling the global singleton `select` which is not great especially because it circumvents match any applied `RegistryProvider`.

I'm planning to do this change in a separate PR

cc @aduth

File: packages/core-data/src/entities.js [hunk 2]
```diff
@@ -46,8 +41,8 @@ async function loadPostTypeEntities() {
  *
  * @return {Promise} Entities promise
  */
-async function loadTaxonomyEntities() {
-	const taxonomies = await apiFetch( { path: '/wp/v2/taxonomies?context=edit' } );
+function * loadTaxonomyEntities() {
```
Comment (tofumatt) [inline packages/core-data/src/entities.js]: Looks like we need an eslint rule for `function*` over `function *`, because this file is the only place you wrote it `function *` and it looks weird to me 😄

File: packages/core-data/src/controls.js [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+ * @param {string} selectorName Selector name.
+ * @param  {Array} args         Selector arguments.
+ *
+ * @return {Object} control descriptor.
+ */
+export function select( selectorName, ...args ) {
+	return {
+		type: 'SELECT',
+		selectorName,
+		args,
```
Comment (tofumatt) [inline packages/core-data/src/controls.js]: It seems weird that this isn't a spread here but is above… does that mean this is doing: `args: …args`? Sometimes I get hazy with ES6 shorthands.

**Review by tofumatt — APPROVED** [review]
This seems to be part of a larger set of changes I've only been partially paying attention to, and while I find generator code less friendly to newbies over the `async/await` syntax I know it can buy us things too…

The code here seems as easier to follow as before, so looks good to me. I might be missing something though, if you wanted to wait for @aduth to have a look I wouldn't be offended 😅

File: packages/core-data/src/controls.js (hunk 3 above)
Comment (youknowriad) [reply packages/core-data/src/controls.js]: In this particular case, it's better to keep it as a separate `args` property, to avoid ambiguity with the others keys (type, selectorName)
